Testclusters: graph

alpar-t opened the pull request:
Convert x-pack  graph to use testClusters

elasticmachine commented:
Pinging @elastic/es-core-features

elasticmachine commented:
Pinging @elastic/es-core-infra

atorok commented on x-pack/plugin/graph/qa/build.gradle:
```diff
@@ -1,17 +0,0 @@
-import org.elasticsearch.gradle.test.RestIntegTestTask
```
This is not necessary as it's part of `RestIntegTestTask`

alpar-t committed:
Testclusters: graph

Convert x-pack  graph to use testClusters

elasticmachine commented:
Pinging @elastic/es-analytics-geo

polyfractal reviewed (approved):
Without knowing much about testClusters, this LGTM :)

/cc @markharwood in case he has something to add

alpar-t committed:
Fix the PR check failures

Looks like this is needed after all, it didn't fail locally because it
only happens after a clean

markharwood commented:
> /cc @markharwood in case he has something to add

Unfamiliar too I'm afraid

mark-vieira commented on x-pack/plugin/graph/qa/with-security/build.gradle:
```diff
@@ -12,25 +13,15 @@ task copyGraphRestTests(type: Copy) {
+integTest.dependsOn copyGraphRestTests
+
+testClusters.integTest {
+  distribution = "DEFAULT"
```
Do we strictly need to use the default distribution here? We should be judicious with this since it's much less expensive to use the integ-test distribution.

atorok replied:
No, but this is what was done with ClusterFormationTasks. 
I'm +1 on using integ-test and adding only the plugins and modules that are strictly needed for the test but  I would look at that separately and discuss it more broadly as all of x-pack is in the same boat.

atorok commented:
@elasticmachine run elasticsearch-ci/bwc

atorok commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into testclusters-graph
